packages/svelte/src/internal/client/runtime.js

```diff
@@ -1420,13 +1420,10 @@ export function user_effect(init) {
 		!apply_component_effect_heuristics
 	);
 	if (apply_component_effect_heuristics) {
-		let effects = /** @type {import('./types.js').ComponentContext} */ (current_component_context)
-			.e;
-		if (effects === null) {
-			effects = /** @type {import('./types.js').ComponentContext} */ (current_component_context).e =
-				[];
-		}
-		effects.push(effect);
+		const context = /** @type {import('./types.js').ComponentContext} */ (
+			current_component_context
+		);
+		(context.e ??= []).push(effect);
```

Comment:
while looking for other examples of this pattern, I found that we often do this, which is quicker, because it avoids a function call for the initial entry:

```js
if (obj.x === null) {
  obj.x = [thing];
} else {
  obj.x.push(thing);
}
```

we could do that here too, not sure if overkill

Reply:
Leaning towards overkill